escrow/state_service.go

```diff
@@ -51,6 +51,29 @@ func (service *PaymentChannelStateService) GetChannelState(context context.Conte
 		return nil, errors.New("only channel signer can get latest channel state")
 	}
 
+	// check if nonce matches with blockchain or not
+	nonceEqual, err := service.channelService.StorageNonceMatchesWithBlockchainNonce(&PaymentChannelKey{ID: channelID})
+	if err != nil {
+		log.WithError(err).Infof("payment data not available in payment storage.")
+	} else if !nonceEqual {
+		// check for payments in the payment storage with current nonce -1, this will happen  cli has issues in claiming process
+		paymentID := fmt.Sprintf("%v/%v", channel.ChannelID, (&big.Int{}).Sub(channel.Nonce, big.NewInt(1)))
```

Comment:
This is the code duplication of `Payment.ID` method. It is unsafe as if we will change the template of a key there is will be two places to update. 

I would propose adding separate function to get payment id from channel id and channel nonce `PaymentID(string channelID, string channelNonce)` into `payment_channel_api.go` module and implement `Payment.ID` using this function.

Reply:
@vsbogd , agreed .
Made this change now